File: Array/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        if (nums == null || nums.length == 0) {
            return new int[] {-1, -1};
        }
        Map<Integer, Integer> visited = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (visited.containsKey(diff)) {
                return new int[] {i, visited.get(diff)};
            }
            visited.put(nums[i], i);
        }
        return new int[] {-1, -1};
    }
}

File: python/hashmap/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i in range(0, len(nums)):
            complement = target - nums[i]
            if complement in seen:
                return [i, seen[complement]]
            else:
                seen[nums[i]] = i
        return None